src/test/kotlin/com/personio/synthetics/builder/api/StepBuilderTest.kt

```diff
@@ -146,6 +150,43 @@ class StepBuilderTest {
         assertEquals(isCritical, result.isCritical)
     }
 
+    @Test
+    fun `build sets step retry defaults if retry method was not called`() {
+        val stepBuilder = StepBuilder("any_name", makeRequestBuilderHappyPathMock())
+        stepBuilder.request { }
+        val result = stepBuilder.build()
+
+        assertEquals(SyntheticsTestOptionsRetry(), result.retry)
+    }
+
+    @Test
+    fun `build sets step retry values if retry method was called`() {
```

Comment:
A bit unclear phrasing, did you mean: builder sets the retry count and interval if the retry method is called?